src/BBox.cpp

```diff
@@ -641,3 +641,33 @@ void BBox::serialize(FileInterface* file) {
 		file->property("mass", mass);
 	}
 }
+
+//These functions are not in the script.cpp file since they drastically increase compile time and memory usage due to heavy template usage.
+void Script::exposeBBox() {
```

Comment:
This is a fine first pass but if these expose functions could be members of their respective classes (ie BBox::exposeToScript() instead of Script::exposeBBox()) it would be way better.